x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/whitelist/InternalSqlScriptUtils.java

```diff
@@ -468,6 +468,10 @@ public static String ucase(String s) {
     // Casting
     //
     public static Object cast(Object value, String typeName) {
-        return DataTypeConversion.convert(value, DataType.fromTypeName(typeName));
+        Object safeDateTimeValue = value;
```

Comment:
It might make sense to rework this as part of `asDateTime` which handles the various types of dates that can be encountered.

Reply:
Yeah, I wasn't 100% happy about the duplicated code, either. I've pushed another approach. Let me know what you think @costin.